pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -176,6 +176,11 @@ func ApplyTaskResults(targets PipelineRunState, resolvedResultRefs ResolvedResul
 			pipelineTask.Params = replaceParamValues(pipelineTask.Params, stringReplacements, arrayReplacements, objectReplacements)
 			if pipelineTask.IsMatrixed() {
 				pipelineTask.Matrix.Params = replaceParamValues(pipelineTask.Matrix.Params, stringReplacements, nil, nil)
+				if pipelineTask.Matrix.HasInclude() {
```

Comment:
this felt to be changes for the reconciler, which doesn't feel quite related to the current PR title/description 🤔